CMakeLists.txt

```
@@ -257,29 +262,30 @@ if(NOT GIT_COMMIT_HASH)
 endif()
 message(STATUS "Git commit: ${GIT_COMMIT_HASH}")
 
-# Get the number of commits on the working branch
+# Check if the worktree is dirty
 execute_process(
-  COMMAND git rev-list --count --first-parent HEAD
+  COMMAND git diff --quiet
+  RESULT_VARIABLE GIT_WORKTREE_DIRTY
   WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-  OUTPUT_VARIABLE GIT_COMMIT_COUNT
-  OUTPUT_STRIP_TRAILING_WHITESPACE
   ERROR_QUIET
 )
-if(NOT GIT_COMMIT_COUNT)
-    set(GIT_COMMIT_COUNT "unknown")
+if(GIT_WORKTREE_DIRTY EQUAL "1")
+  set(GIT_COMMIT_HASH "${GIT_COMMIT_HASH} (modified)")
 endif()
 
-# Check if the worktree is dirty
+set(ENV{TZ} UTC)
 execute_process(
-  COMMAND git diff --quiet
-  RESULT_VARIABLE GIT_WORKTREE_DIRTY
+  COMMAND git show --quiet --date=format-local:%Y-%m-%d_%H-%M-%S --format=%cd
```

```suggestion
  COMMAND git show --quiet --date=format-local:%Y-%m-%d-%H-%M-%S --format=%cd
```
underscores are not allowed in the version

---

okay